addon/components/citation-plugin/citation-insert.ts

```diff
@@ -62,28 +63,32 @@ export default class EditorPluginsCitationInsertComponent extends Component<Args
       return true;
     }
     const { selection } = this.controller.mainEditorState;
-    if (this.config.type === 'ranges') {
-      const ranges = this.config.activeInRanges(
-        this.controller.mainEditorState,
-      );
+    const config = this.config;
+    if (config.type === 'ranges') {
+      const ranges = config.activeInRanges(this.controller.mainEditorState);
       for (const range of ranges) {
         if (selection.from > range[0] && selection.from < range[1]) {
           return false;
         }
       }
       return true;
     } else {
-      const nodeTypes = this.config.activeInNodeTypes(
-        this.controller.schema,
-        this.controller.mainEditorState,
-      );
-      // if the doc node is included, the button should always be active
-      // the findParentNodeOfType util we import does NOT consider the doc node
-      // in its search.
-      if (nodeTypes.has(this.controller.schema.nodes.doc)) {
+      let condition: (node: PNode) => boolean;
+      if ('activeInNodeTypes' in config) {
+        const nodeTypes = config.activeInNodeTypes(
+          this.controller.schema,
+          this.controller.mainEditorState,
+        );
+        condition = (node) => nodeTypes.has(node.type);
+      } else {
+        condition = (node) =>
+          config.activeInNode(node, this.controller.mainEditorState);
```

Comment:
We probably typed it this way, but it would probably be best to handle the absence of any config, regardless of the config type, by enabling in the whole doc

same for the absence of the 'type' config

Reply:
Makes sense indeed, I'll adjust it.

Reply:
is adjusted :)